src/axom/slic/tests/slic_lumberjack_parallel.cpp

```diff
@@ -0,0 +1,395 @@
+#include "axom/slic/interface/slic_macros.hpp"
+#include "axom/slic/streams/LumberjackStream.hpp"
+
+#include "gtest/gtest.h"
+
+#include <string>
+#include <sstream>
+
+#include "mpi.h"
+
+// namespace alias
+namespace slic = axom::slic;
+
+namespace axom
+{
+namespace slic
+{
+namespace internal
+{
+std::ostringstream test_stream;
+
+bool is_stream_empty() { return test_stream.str().empty(); }
+
+void clear_stream()
+{
+  test_stream.clear();
+  test_stream.str("");
+  EXPECT_TRUE(is_stream_empty());
+}
+
+}  // namespace internal
+}  // namespace slic
+}  // namespace axom
+
+//------------------------------------------------------------------------------
+// HELPER METHODS
+//------------------------------------------------------------------------------
+namespace
+{
+void check_level(const std::string& msg, const std::string& expected_level)
+{
+  EXPECT_FALSE(msg.empty());
+
+  // extract level
+  size_t start = msg.find("[") + 1;
+  size_t end = expected_level.length();
+  std::string level = msg.substr(start, end);
+
+  // check with expected
+  EXPECT_FALSE(level.empty());
+  EXPECT_EQ(level, expected_level);
+}
+
+//------------------------------------------------------------------------------
+void check_msg(const std::string& msg, const std::string& expected_message)
+{
+  EXPECT_FALSE(msg.empty());
+
+  // extract message
+  size_t start = msg.find(";;") + 2;
+  size_t end = expected_message.length();
+  std::string mymsg = msg.substr(start, end);
+
+  EXPECT_FALSE(mymsg.empty());
+  EXPECT_EQ(mymsg, expected_message);
+}
+
+//------------------------------------------------------------------------------
+void check_file(const std::string& msg)
+{
+  EXPECT_FALSE(msg.empty());
+
+  const std::string EXPECTED_FILE = std::string(__FILE__);
+
+  // extract filename
+  size_t start = msg.find("@@") + 2;
+  size_t end = EXPECTED_FILE.length();
+  std::string myfile = msg.substr(start, end);
+
+  EXPECT_FALSE(myfile.empty());
+  EXPECT_EQ(myfile, std::string(__FILE__));
+}
+
+//------------------------------------------------------------------------------
+void check_line(const std::string& msg, int expected_line)
+{
+  EXPECT_FALSE(msg.empty());
+
+  // extract line
+  size_t start = msg.rfind("@@") + 2;
+  std::string l = msg.substr(start);
+  const int line = std::stoi(l);
+  EXPECT_EQ(line, expected_line);
+}
+
+//------------------------------------------------------------------------------
+bool has_aborted = false;
+
+void custom_abort_function() { has_aborted = true; }
+void reset_state()
+{
+  axom::slic::internal::clear_stream();
+  has_aborted = false;
+}
+
+}  // end anonymous namespace
+
+//------------------------------------------------------------------------------
+// UNIT TESTS
+//------------------------------------------------------------------------------
+//------------------------------------------------------------------------------
+TEST(slic_lumberjack_parallel, test_abort_error_macros)
```

Comment:
If this is just a copy of the parallel tests, we could move this to a templated testing suite...

Reply:
> If this is just a copy of the parallel tests, we could move this to a templated testing suite...

It is, the main difference being these parallel lumberjack tests are only defined if axom is built with lumberjack. 
If I combine the two, the parallel tests will be defined only if axom is built with slic and lumberjack. 
I think it'll make future maintenance easier to combine them, just noting this here so I remember later.